singer_sdk/plugin_base.py

```diff
@@ -171,14 +195,36 @@ def _get_package_version(package: str) -> str:
             version = "[could not be detected]"
         return version
 
+    @staticmethod
+    def _get_supported_python_versions(package: str) -> list[str] | None:
+        """Return the supported Python versions.
+
+        Args:
+            package: The package name.
+
+        Returns:
+            A list of supported Python versions.
+        """
+        try:
+            supported_python_versions = []
+            package_metadata = metadata.metadata(package)
+            reported_python_versions = SpecifierSet(package_metadata["Requires-Python"])
```

Comment:
Any idea why, despite the updated test, Codecov is not seeing this line as covered?

Reply:
Our tests for `--about` test the formatters with example data - we don't have a test for how the data is sourced. I suspect this is because of the complexity of installing example packages to test against. Will look at how easy this is to do with pytest as I agree it would be good to test the sourcing side 👍

Reply:
We do have https://github.com/meltano/sdk/issues/915 so in the future we could make the samples into installable packages and this behavior could be tested with them.

Whether we decide to distribute them separately is a different question, imo.